src/main/java/com/meow/Parser.java

```diff
@@ -0,0 +1,298 @@
+package com.meow;
+import java.util.Scanner;
+
+import com.meow.com.tasks.Task;
+import com.meow.com.tasks.Todo;
+import com.meow.com.tasks.Event;
+
+import com.meow.com.tasks.Deadline;
+public class Parser {
+    private Scanner scanner;
+    private TaskList taskList;
+    public enum Commands {
+        TODO, DEADLINE, EVENT, LIST, DELETE, MARK, UNMARK, BYE,
+        FIND
+    }
+
+    /**
+     * Constructor for Parser class
+     * @param TaskList
+     * @return void
+     */
+    public Parser(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    /*
+     * Validates the command input
+     * @param String inputType
+     * @return String
+     */
+    public String commandValidation(String inputType) throws Meowception {
+        try {         
+            String[] parts = inputType.split(" ");
+            Commands userCommand = Commands.valueOf(parts[0].toUpperCase());
+            return outputTask(userCommand, inputType);
+        } catch (StringIndexOutOfBoundsException e) {
+            throw new Meowception("404");
+        } catch (IllegalArgumentException e) {
+            throw new Meowception("001");
+        }
+    }
+
+    /*
+     * Outputs the task based on the command
+     * @param Commands cmd
+     * @param String inputType
+     * @return String
+     */
+    private String outputTask(Commands cmd,String inputType) throws Meowception {
```

Comment:
Should there be a space between the parameters? 

```suggestion
    private String outputTask(Commands cmd, String inputType) throws Meowception {
```